### PR TITLE
fix(multi-env): improve resource group permission error message

### PR DESCRIPTION
**Pull request by a1exwang**
When selecting resource group before provision, and if the user does not have permission to acess current selected subscription, it will display a more user friendly error message.
image

https://msazure.visualstudio.com/Microsoft%20Teams%20Extensibility/_workitems/edit/12545873

### CODE REVIEW

**Comment by a1exwang** [comment]
E2E TEST: https://github.com/OfficeDev/TeamsFx/blob/dev/packages/cli/tests/e2e/multienv/TestRemoteHappyPath.tests.ts

File: packages/fx-core/src/plugins/solution/fx-solution/commonQuestions.ts [hunk 1]
```diff
@@ -671,24 +691,68 @@ async function createNewResourceGroup(
...
+      message: restError.message,
+    });
+    return err(
+      returnUserError(
+        new Error(
+          `Failed to check the existence of resource group '${resourceGroupName}' in subscription '${subscriptionInfoString}', raw error: '${rawErrorString}'`
+        ),
+        SolutionSource,
+        SolutionError.FailedToCheckResourceGroupExistence
+      )
```
Comment (dooriya) [inline packages/fx-core/src/plugins/solution/fx-solution/commonQuestions.ts:728]: We have several `UserError` related to the resource group check, is that possible to wrap/abstract these kinds of `UserError` for better error handling/monitoring, I remember Huajie has talked about how to define the Error in Teams toolkit before.

File: packages/fx-core/src/plugins/solution/fx-solution/commonQuestions.ts [hunk 2]
```diff
@@ -671,24 +691,68 @@ async function createNewResourceGroup(
...
-    // Wrap the error to prevent showing empty error message to the users.
     let error;
-    const errorMessage = `Failed to check the existence of the resource group '${resourceGroupName}', error: '${e}'`;
-    if (e instanceof Error) {
+    if (e instanceof RestError) {
+      return handleRestError(e, resourceGroupName, subscriptionId, subscriptionName);
+    } else if (e instanceof Error) {
       // reuse the original error object to prevent losing the stack info
-      e.message = errorMessage;
+      e.message = `Failed to check the existence of resource group '${resourceGroupName}', error: '${e}'`;
```
Comment (qinezh) [inline packages/fx-core/src/plugins/solution/fx-solution/commonQuestions.ts:748]: > `error: '${e}'`
do we need to stringify the object as well here?

Comment (a1exwang) [reply packages/fx-core/src/plugins/solution/fx-solution/commonQuestions.ts:748]: The reason here is for some other types of non-RestError (like network error), the user can know:
1. The error is related to checking resource group
2. The true reason of the error so he can take action or report issue.

So I wrap these two pieces of information into this error message.

File: packages/fx-core/src/plugins/solution/fx-solution/commonQuestions.ts (hunk 1 above)
Comment (a1exwang) [reply packages/fx-core/src/plugins/solution/fx-solution/commonQuestions.ts:728]: It makes sense. I will refactor it.

File: packages/fx-core/src/plugins/solution/fx-solution/constants.ts [hunk 3]
```diff
@@ -234,3 +234,51 @@ export enum SolutionTelemetrySuccess {
...
+      subscriptionId + (subscriptionName.length > 0 ? `(${subscriptionName})` : "");
+    super(
+      new.target.name,
+      `Unauthorized to check the existence of resource group '${resourceGroupName}' in subscription '${subscriptionInfoString}'. Please check your Azure subscription.`,
+      SolutionSource
+    );
+  }
+}
+
+export class FailedToCheckResourceGroupExistence extends UserError {
```
Comment (dooriya) [inline packages/fx-core/src/plugins/solution/fx-solution/constants.ts:250]: nit: can we name it with `Error` suffix?

**Review by qinezh — APPROVED** [review]

**Review by dooriya — APPROVED** [review]

**Review by jayzhang — APPROVED** [review]

**Pull request merged**